packages/chain-adapters/src/evm/avalanche/AvalancheChainAdapter.ts

```diff
@@ -0,0 +1,206 @@
+      chainId: this.chainId,
+      rpcUrl: args.rpcUrl
+    })
+  }
+
+  getType(): KnownChainIds.AvalancheMainnet {
+    return KnownChainIds.AvalancheMainnet
+  }
+
+  getFeeAssetId(): AssetId {
+    return avalancheAssetId
+  }
+
+  async buildSendTransaction(tx: BuildSendTxInput<KnownChainIds.AvalancheMainnet>): Promise<{
+    txToSign: ETHSignTx
+  }> {
+    try {
+      const { to, wallet, bip44Params = ChainAdapter.defaultBIP44Params, sendMax = false } = tx
+      const { erc20ContractAddress, gasPrice, gasLimit, maxFeePerGas, maxPriorityFeePerGas } =
+        tx.chainSpecific
+
+      if (!tx.to) throw new Error('AvalancheChainAdapter: to is required')
+      if (!tx.value) throw new Error('AvalancheChainAdapter: value is required')
+
+      const destAddress = erc20ContractAddress ?? to
+
+      const from = await this.getAddress({ bip44Params, wallet })
+      const account = await this.getAccount(from)
+
+      const isErc20Send = !!erc20ContractAddress
+
+      if (sendMax) {
+        if (isErc20Send) {
+          const erc20Balance = account?.chainSpecific?.tokens?.find((token) => {
+            return fromAssetId(token.assetId).assetReference === erc20ContractAddress.toLowerCase()
+          })?.balance
+          if (!erc20Balance) throw new Error('no balance')
+          tx.value = erc20Balance
+        } else {
+          if (bnOrZero(account.balance).isZero()) throw new Error('no balance')
+
+          // (The type system guarantees that either maxFeePerGas or gasPrice will be undefined, but not both)
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          const fee = bnOrZero((maxFeePerGas ?? gasPrice)!).times(bnOrZero(gasLimit))
+          tx.value = bnOrZero(account.balance).minus(fee).toString()
+        }
+      }
+      const data = await getErc20Data(to, tx?.value, erc20ContractAddress)
+
+      const fees = ((): Fees => {
+        if (maxFeePerGas && maxPriorityFeePerGas) {
+          return {
+            maxFeePerGas: numberToHex(maxFeePerGas),
+            maxPriorityFeePerGas: numberToHex(maxPriorityFeePerGas)
+          }
+        }
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return { gasPrice: numberToHex(tx.chainSpecific.gasPrice!) }
+      })()
+
+      const txToSign: ETHSignTx = {
+        addressNList: bip32ToAddressNList(toPath(bip44Params)),
+        value: numberToHex(isErc20Send ? '0' : tx?.value),
+        to: destAddress,
+        chainId: 43114, // TODO: implement for multiple chains
+        data,
+        nonce: numberToHex(account.chainSpecific.nonce),
+        gasLimit: numberToHex(gasLimit),
+        ...fees
+      }
+      return { txToSign }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async getGasFeeData(): Promise<GasFeeDataEstimate> {
+    const feeData = (await this.providers.http.getGasFees()).data
+
+    const normalizationConstants = {
```

Comment:
magic +/- 20% fee diff for slow and fast, we can iterate to dial in more accurate fees. this differs from eth where we were using 0x gas api for fees